libsolidity/analysis/ControlFlowBuilder.cpp

```diff
@@ -409,7 +409,7 @@ bool ControlFlowBuilder::visit(InlineAssembly const& _inlineAssembly)
 void ControlFlowBuilder::visit(yul::Statement const& _statement)
 {
 	solAssert(m_currentNode && m_inlineAssembly, "");
-	m_currentNode->location = langutil::SourceLocation::smallestCovering(m_currentNode->location, locationOf(_statement));
+	m_currentNode->location = langutil::SourceLocation::smallestCovering(m_currentNode->location, nativeLocationOf(_statement));
```

Comment:
Changing this to `originLocationOf()` does not break any tests so I think we don't have coverage for this in case where Yul source has debug info annotations.

Reply:
Just to check: `nativeLocationOf(_statement)` always returns a "proper" source location, no matter in which mode the parser was used, right?

What I'm a bit worried about is the possibility of the location being the same when we use `appendAssembly` in the old code generator and supply two different snippets (that would not have a different name) but two yul ast items have the same native locations inside these snippets.

IIRC, `location` is not only used for informational purposes, but to check equality in the source - is this the case here?

Reply:
Ah wait, this is only about inline assembly here (and not compiler-generated assembly for helper routines), so native location and origin location should be the same anyway. Maybe assert that?

Reply:
> Just to check: `nativeLocationOf(_statement)` always returns a "proper" source location, no matter in which mode the parser was used, right?

> IIRC, `location` is not only used for informational purposes, but to check equality in the source - is this the case here?

Yeah, it looks like it's using it to disambiguate the stuff it adds to `variableOccurrences`.

> What I'm a bit worried about is the possibility of the location being the same when we use `appendAssembly` in the old code generator and supply two different snippets (that would not have a different name) but two yul ast items have the same native locations inside these snippets.

I don't really understand this. We're not storing snippets in debug info or passing them around. Or do you mean the inline assembly snippets we compile with the location override? I'm also not sure what the consequences of these locations being the same would be.

> Ah wait, this is only about inline assembly here (and not compiler-generated assembly for helper routines), so native location and origin location should be the same anyway. Maybe assert that?

Ah, you're right. It's in `libsolidity` so it must be. Assertions added.